### PR TITLE
feature: update README.md

### PR DESCRIPTION
**Pull request by yizhibian**
as title

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,56 @@
...
+make build
+make run 
+```
+
+## Test
+
+This test case is based on a local Envoy. Run it with the example config file.
+
+```bash
+go test test/e2e_authorized_test.go test/common.go
```
Comment (spacewander) [inline README.md]: Should be `go test ./test`? Would be better to wrap it in `make test` for consistency.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,56 @@
 # envoy-go-basic-auth
-Basic Auth based on Envoy Golang http filter
+
+This is a simple basic auth filter for envoy written in go. Only requests that pass the configuration authentication will be proxied to the upstream service.
+
+## Status
+
+This is under active development and is not ready for production use.
```
Comment (spacewander) [inline README.md]: Is this plugin still under development?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,56 @@
...
+credentials := Basic base64(username:password)
+```
+
+An example of the `Authorization` header is as follows (`Zm9vOmJhcg==`, which is the base64-encoded value of `foo:bar`):
+
+```Plaintext
+Authorization: Basic Zm9vOmJhcg==
+```
+
+Configure your envoy.yaml to set pairs of username and password.
```
Comment (spacewander) [inline README.md]: The `make run` will use the `envoy. yaml` in this project. Maybe we can provide a link here so people can know which configuration will be used.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,56 @@
...
+
+This is a simple basic auth filter for envoy written in go. Only requests that pass the configuration authentication will be proxied to the upstream service.
+
+## Status
+
+This is under active development and is not ready for production use.
+
+## Usage
+
+The client set credentials in `Authorization` header in the following format:
```
Comment (spacewander) [inline README.md]:
```suggestion
The client sets credentials in `Authorization` header in the following format:
```

File: README.md (hunk 1 above)
Comment (yizhibian) [reply README.md]: My fault, fix it now.

File: README.md (hunk 2 above)
Comment (yizhibian) [reply README.md]: Sry, can I change it to 'The plugin has been completed and will be updated in the future.'?

File: README.md (hunk 3 above)
Comment (yizhibian) [reply README.md]: sure, fix it now.

File: README.md (hunk 2 above)
Comment (spacewander) [reply README.md]: LGTM

File: README.md [hunk 5]
```diff
@@ -1,2 +1,56 @@
...
+make build
+make run 
+```
+
+## Test
+
+This test case is based on a local Envoy. Run it with the example config file.
+
+```bash
+make test
```
Comment (spacewander) [inline README.md:55]: Let's update the CI to use `make test` together

Comment (yizhibian) [reply README.md:55]: Yep, update it now.

File: README.md [hunk 6]
```diff
@@ -4,7 +4,7 @@ This is a simple basic auth filter for envoy written in go. Only requests that p
 
 ## Status
 
-This is under active development and is not ready for production use.
+The plugin has been completed and will be updated in the future.'
```
Comment (spacewander) [inline README.md]:
```suggestion
The plugin has been completed and may be updated in the future.
```

**Review by spacewander — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**